apps/emqx/src/emqx_frame.erl

```diff
@@ -260,42 +271,52 @@ parse_packet(
     FrameBin,
     #{strict_mode := StrictMode}
 ) ->
-    {ProtoName, Rest} = parse_utf8_string(FrameBin, StrictMode),
-    <<BridgeTag:4, ProtoVer:4, Rest1/binary>> = Rest,
-    % Note: Crash when reserved flag doesn't equal to 0, there is no strict
-    % compliance with the MQTT5.0.
-    <<UsernameFlag:1, PasswordFlag:1, WillRetain:1, WillQoS:2, WillFlag:1, CleanStart:1, 0:1,
-        KeepAlive:16/big, Rest2/binary>> = Rest1,
-
-    {Properties, Rest3} = parse_properties(Rest2, ProtoVer, StrictMode),
-    {ClientId, Rest4} = parse_utf8_string(Rest3, StrictMode),
-    ConnPacket = #mqtt_packet_connect{
-        proto_name = ProtoName,
-        proto_ver = ProtoVer,
-        is_bridge = (BridgeTag =:= 8),
-        clean_start = bool(CleanStart),
-        will_flag = bool(WillFlag),
-        will_qos = WillQoS,
-        will_retain = bool(WillRetain),
-        keepalive = KeepAlive,
-        properties = Properties,
-        clientid = ClientId
-    },
-    {ConnPacket1, Rest5} = parse_will_message(ConnPacket, Rest4, StrictMode),
-    {Username, Rest6} = parse_utf8_string(Rest5, StrictMode, bool(UsernameFlag)),
-    {Password, <<>>} = parse_utf8_string(Rest6, StrictMode, bool(PasswordFlag)),
-    ConnPacket1#mqtt_packet_connect{username = Username, password = Password};
+    case parse_utf8_string(FrameBin, StrictMode) of
```

Comment:
refactor this clause to a smaller function.

Reply:
I think there is small enough, I don't know how to simplify it.